src/WebJobs.Extensions.DurableTask/IPlatformInformationService.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Interface for accessing the AppService plan information,
+    /// the OS, and user-facing PL.
+    ///
+    /// Note: The functionality is currently limited, but will grow
+    /// along with the pursuit of more platform-specific defaults.
+    /// </summary>
+    public interface IPlatformInformationService
```

Comment:
I think we may be able to clean this up a little bit in a way that would appeal to the functional programmer in you.

Imagine we have `IPlatformInformationService` have a single method `PlatformInformation GetPlatformInformation()`.

Then you can have various subclasses of PlatformInformation, such as AppServiceWindows, AppServiceWindowsConsumption, AppServiceLinux, AppServiceLinuxConsumption, Unknown, etc. 

That type hierarchy alone will allow you to eliminate many of these methods.

Then, you can put the information that is only specific to windows or linux only on the appropriate subclasses.

You can then use pattern matching for these various cases, as opposed to a series of if-statements.

Overall, this is more of a matter of preference if you prefer a sequence of if-statements vs a type based approach + pattern matching, but given your background in PL, I thought I would mention this option.

Reply:
Haha, I've been nerd-snipped.
Yeah I see what you mean and I do like it. I can certainly see this being a clean solution for serving OS-specific information. I suppose I could also bundle PL-specific data in there, I still need to think about that.

I'll get back to this. I think there's still some contention about how many settings to optimize and what values to give them so let me focus on that first since we're time-constrained. There will definitely be a follow-up PR to this for OOProc and we can use that to finalize the design of this utility if we don't get to it now.

Reply:
I'll opt to postpone this refactor until the OOProc performance tuning PR